kevm-pyk/src/kevm_pyk/cli.py

```diff
@@ -32,382 +48,758 @@ def node_id_like(s: str) -> NodeIdLike:
         return s
 
 
-class KEVMCLIArgs(KCLIArgs):
-    @cached_property
-    def target_args(self) -> ArgumentParser:
-        args = ArgumentParser(add_help=False)
-        args.add_argument('--target', choices=['llvm', 'haskell', 'haskell-standalone', 'foundry'])
-        return args
+def generate_command_options(args: dict[str, Any]) -> LoggingOptions:
+    command = args['command'].lower()
+    match command:
+        case 'version':
+            return VersionOptions(args)
+        case 'kompile-spec':
+            return KompileSpecOptions(args)
+        case 'prove':
+            return ProveOptions(args)
+        case 'prune-proof':
+            return PruneProofOptions(args)
+        case 'prove-legacy':
+            return ProveLegacyOptions(args)
+        case 'view-kcfg':
+            return ViewKCFGOptions(args)
+        case 'show-kcfg':
+            return ShowKCFGOptions(args)
+        case 'run':
+            return RunOptions(args)
+        case 'kast':
+            return KastOptions(args)
+        case _:
+            raise ValueError('Unrecognized command.')
```

Comment:
This dispatch feels a bit weird to me as well. If we had one single `KEVMCLI` class, that inherited all the other classes and put together all the options, then we would know that we had handled enough of the options. As it stands, this must be kept in sync with the other place where all the options are put together.